src/conn.rs

```diff
@@ -219,6 +219,7 @@ pub struct InProgress<'a, 'c> {
     use_caching: bool,
     tx_ids: AccumulatedTxids,
     tx_observer: &'a Mutex<TxObservationService>,
```

Comment:
OK, this makes sense. I got into my lifetime problems by trying to keep the `TxObservationService` inside the watcher. Removing it makes everything a lot simpler.